opentracing/src/main/java/opentracing/TraceContextSource.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+import java.util.Map;
+
+/**
+ * Long-lived interface that knows how to create a root {@linkplain TraceContext} and encode/decode
+ * any other.
+ */
+public interface TraceContextSource {
```

Comment:
I'm good with not-having `Tracer` inherit from this, but shouldn't it (`Tracer`, I mean) have a `TraceContextSource source()` or `TraceContextSource traceContextSource()` method?